MIP/mip-34/README.md

```diff
@@ -0,0 +1,261 @@
+
+> [!NOTE]
+> Until a better definition arises we consider _**confirmation**_ to be defined as _L2-finality_ (i.e. _L2-confirmation_).
+
+**L2-confirmation**. FFS aims to _confirm_ the validity of each produced L2-block, at every L2-block height. 
+> [!IMPORTANT]
+> If we confirm each successor block before adding it to the (confirmed) L2-chain, there cannot be any fork, except if the sequencer would provide equivocating sequencer-batches for a given height AND there is a sufficiently strong Byzantine attack on the confirmation process.
+
+If the validators can attest blocks quickly and make their attestations available to third-parties, we have a fast confirmation mechanism supported by crypto-econonimic security, the  level of which depends on what is at stake for the confirmation of a block.
+
+
+**Postconfirmation**. At certain intervals confirmation will also be achieved on L1. The L1 contract will verify the satisfaction of the super-majority critera. This provides an L1-protected _postconfirmation_ that the block (or a batch of blocks, which we call _block range_) is confirmed. This additional anchoring mechanism increases the security of the L2-confirmation as it locks in the L2-confirmation, reduces the risk of long range attacks and provides a way to slash validators that have attested for invalid blocks.
+
+**Slashing**. The security of the mechanism relies on a PoS protocol. Each validator has to stake some assets, and if they are malicious they _should_ be slashed.
+The condition for slashing may be met by several quiteria, and not all slashing conditions may be used:
+- equivocate (send a different vote to different validators or users)
+- vote :white_check_mark: for an invalid block 
+- vote :x: for a valid block
+
+### Main challenges
+
+To achieve crypto-economically secured fast finality, we need to solve the following problems:
+
+1. design a _staking_ mechanism for the validators to stake assets, distribute rewards and manage slashing
+1. _define and verify_ the threshold (e.g. 2/3 of validators attest :white_check_mark:) for L2-finality
+1. _communicate_ the L2-finality status.
+
+In addition the following may require separate discussion, as it is a different procedure (postconfirmations are handled in smart contracts on L1, whereas L2-confirmations are handled off-chain)
+
+4. _define and verify_ the threshold (e.g. 2/3 of validators attest :white_check_mark:) for postconfirmation.
+
+
+### Components
+
+#### Staking (addresses 1. and 2.)
+
+The staking mechanism is implemented in a contract on L1.
+This contract provides the following functionalities:
+
+- join: a new validator can join the set of validators by staking some assets
+- exit: a validator exits and  get their stakes back
+- vote: receive a vote or a set of votes, verify the integrity of the votes (signatures) and the minimum threshold (e.g. 2/3)
```

Comment:
Later today, I was going to introduce a gloss for consensus that I'm sure will stir debate. But, I don't think we should "e.g.-ing" the $\frac{2}{3}$. 

Also, use a LaTeX frac here.

Reply:
Here is a starting point: 

> Nodes in the network participate in a collaborative protocol (consensus) to agree on whether the block should be added to the blockchain.
> https://dl.acm.org/doi/10.1145/3318041.3355458